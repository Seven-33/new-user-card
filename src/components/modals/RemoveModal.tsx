import { Button, Modal } from "antd";

export type RemoveModalProps = {
  remove: () => void;
  cancel: () => void;
  isOpen: boolean;
};

const RemoveModal = ({ remove, cancel, isOpen }: RemoveModalProps) => {
  return (
    <>
      <Modal
        title="Delete"
        visible={isOpen}
        onOk={remove}
        onCancel={cancel}
        footer={[
          <Button key="confirm" type="primary" onClick={remove}>
            Confirm
          </Button>,
          <Button key="cancel" onClick={cancel}>
            Cancel
          </Button>,
        ]}
      >
        <p>I want to remove this user.</p>
      </Modal>
    </>
  );
};

export default RemoveModal;
