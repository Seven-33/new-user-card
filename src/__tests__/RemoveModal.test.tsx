import RemoveModal, {
  RemoveModalProps,
} from "../components/modals/RemoveModal";
import { render } from "@testing-library/react";

const makeSut = (props: RemoveModalProps) => {
  return render(<RemoveModal {...props} />);
};

describe("This suit is to test the RemoveModal component", () => {
  test("Sanpshot of RemoveModal", () => {
    const { asFragment } = makeSut({
      remove: jest.fn(),
      cancel: jest.fn(),
      isOpen: true,
    });
    expect(asFragment()).toMatchSnapshot();
  });

  test("Should render correctly", () => {
    const { getByText } = makeSut({
      remove: jest.fn(),
      cancel: jest.fn(),
      isOpen: true,
    });
    expect(getByText(/I want to remove this user/)).toBeInTheDocument();
  });
});
