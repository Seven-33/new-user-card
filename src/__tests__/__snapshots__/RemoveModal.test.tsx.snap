// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the RemoveModal component Sanpshot of RemoveModal 1`] = `<DocumentFragment />`;
